pkg/cloudprovider/aws/object_storage_adapter.go

```diff
@@ -27,16 +28,24 @@ import (
 var _ cloudprovider.ObjectStorageAdapter = &objectStorageAdapter{}
 
 type objectStorageAdapter struct {
-	s3 *s3.S3
+	s3       *s3.S3
+	kmsKeyID string
 }
 
 func (op *objectStorageAdapter) PutObject(bucket string, key string, body io.ReadSeeker) error {
+
 	req := &s3.PutObjectInput{
 		Bucket: &bucket,
 		Key:    &key,
 		Body:   body,
 	}
 
+	// if kmsKeyID is not empty, enable "aws:kms" encryption and amend the key ID
```

Comment:
I'm not sure "amend" is the correct word. Maybe just end the comment at "encryption"?